src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/stop/TransportStopTransformAction.kt

```diff
@@ -0,0 +1,159 @@
+                    }
+                    if (metadata == null) {
+                        // If there is no metadata there is nothing to stop, proceed to disable job
+                        updateTransformJob(transform, request, actionListener)
+                    } else {
+                        updateTransformMetadata(transform, metadata, request, actionListener)
+                    }
+                }
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    /**
+     * Updates the transform metadata if required.
+     *
+     * The update is dependent on what the current [TransformMetadata.status] is.
+     * When stopping a transform that is in INIT, STARTED, or STOPPED we will update to STOPPED.
+     * When the transform is in FINISHED or FAILED it will remain as that status.
+     */
+    private fun updateTransformMetadata(
+        transform: Transform,
+        metadata: TransformMetadata,
+        request: StopTransformRequest,
+        actionListener: ActionListener<AcknowledgedResponse>
+    ) {
+        val now = Instant.now().toEpochMilli()
+        val updatedStatus = when (metadata.status) {
+            TransformMetadata.Status.STARTED, TransformMetadata.Status.INIT, TransformMetadata.Status.STOPPED -> TransformMetadata.Status.STOPPED
+            TransformMetadata.Status.FINISHED, TransformMetadata.Status.FAILED -> metadata.status
+        }
+
+        val updateRequest = UpdateRequest(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX, transform.metadataId)
+            .doc(mapOf(TransformMetadata.TRANSFORM_METADATA_TYPE to mapOf(TransformMetadata.STATUS_FIELD to updatedStatus.type,
+                TransformMetadata.LAST_UPDATED_AT_FIELD to now)))
+        client.update(updateRequest, object : ActionListener<UpdateResponse> {
+            override fun onResponse(response: UpdateResponse) {
+                if (response.result == DocWriteResponse.Result.UPDATED) {
+                    updateTransformJob(transform, request, actionListener)
+                } else {
+                    actionListener.onResponse(AcknowledgedResponse(false))
+                }
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun updateTransformJob(transform: Transform, request: StopTransformRequest, actionListener: ActionListener<AcknowledgedResponse>) {
+        val now = Instant.now().toEpochMilli()
+        request.index(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX).setIfSeqNo(transform.seqNo).setIfPrimaryTerm(transform.primaryTerm)
+            .doc(mapOf(Transform.TRANSFORM_TYPE to mapOf(Transform.ENABLED_FIELD to false,
+                Transform.ENABLED_AT_FIELD to null, Transform.ENABLED_AT_FIELD to now)))
```

Comment:
duplication of Transform.ENABLED_AT_FIELD

Reply:
proper field corrected to UPDATED_AT_FIELD